includes/options/views/class-amp-setup-wizard-submenu-page.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * AMP setup wizard page.
+ *
+ * @package AMP
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+
+/**
+ * AMP setup wizard submenu page class.
+ *
+ * @since @todo NEW_ONBOARDING_RELEASE_VERSION
+ */
+class AMP_Setup_Wizard_Submenu_Page {
+	/**
+	 * Handle for JS file.
+	 *
+	 * @var string
+	 */
+	const JS_HANDLE = 'amp-setup';
+
+	/**
+	 * HTML ID for the app root element.
+	 *
+	 * @var string
+	 */
+	const APP_ROOT_ID = 'amp-setup';
+
+	/**
+	 * The minimum core WP version whose core assets are compatibile with the setup feature.
+	 *
+	 * @var float
+	 */
+	const TARGET_WP_VERSION = 5.4;
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'override_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Renders the setup screen markup.
+	 */
+	public function render() {
+		?>
+			<div id="<?php echo esc_attr( static::APP_ROOT_ID ); ?>"></div>
+		<?php
+	}
+
+	/**
+	 * Provides the setup screen handle.
+	 *
+	 * @return string
+	 */
+	public function screen_handle() {
+		return sprintf( 'amp_page_%s', AMP_Setup_Wizard_Submenu::SCREEN_ID );
+	}
+
+	/**
+	 * Registers single script.
+	 *
+	 * @param string  $handle  Script handle.
+	 * @param boolean $enqueue Whether to enqueue the asset immediately. Default false.
+	 */
+	public function add_setup_script( $handle, $enqueue = false ) {
+		$asset = $this->get_asset( $handle, sprintf( '%s/assets/js/%s.asset.php', AMP__DIR__, $handle ) );
```

Comment:
I'd prefer the following notation here, to avoid `sprintf()` and make it more readable:
```suggestion
		$asset = $this->get_asset( $handle, AMP__DIR__ . "/assets/js/{$handle}.asset.php" );
```